src/main/scala/com/workday/warp/utils/AnnotationReader.scala

```diff
@@ -43,18 +40,7 @@ object AnnotationReader extends StackTraceFilter {
    */
   protected def getWarpTestMethod(testId: String): Option[Method] = {
     val methodName: String = testId drop testId.lastIndexOf('.') + 1
-
-    // get an Option containing the Class based on the current value of testID
-    getWarpTestClass(testId) flatMap { clazz =>
-      try {
-        Option(clazz.getMethod(methodName, new Array[Class[_]](0): _*))
-      }
-      catch {
-        // return None if we can't locate the method
-        case e: NoSuchMethodException =>
-          None
-      }
-    }
+    this.getWarpTestClass(testId) flatMap { clazz => Try(Option(clazz.getMethod(methodName, new Array[Class[_]](0): _*))).toOption.flatten }
```

Comment:
what is this sorcery

Reply:
kinda confusing, but we are looking up a `Method` based on its fully qualified signature. 

first we look up the `Class`, then we try to look up its method with the given name.

this weird `:_*` syntax (I read it as "exploding an array") is interop with java varags.

check out the signature of `Class#getMethod` (from Class.java):
```
public Method getMethod(String name, Class<?>... parameterTypes)
```

we know that the method we are trying to look up has 0 parameters (because it is a junit test method), so we construct an empty array and "explode" it to pass as varargs.

i'll make it a separate val or use named method parameters so this is more clear.